src/common/base/Arena.h

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <type_traits>
+
+#include "common/base/Logging.h"
+#include "common/cpp/helpers.h"
+
+namespace nebula {
+
+// MT-unsafe arena allocator
+// It's optimized for many small objects construct/destruct
+class Arena : public cpp::NonCopyable, cpp::NonMovable {
+ public:
+  ~Arena() {
+    while (currentChunk_ != nullptr) {
+      auto *prev = currentChunk_->prev;
+      delete[] currentChunk_;
+      currentChunk_ = prev;
+    }
+#ifndef NDEBUG
+    allocatedSize_ = 0;
+#endif
+    availableSize_ = 0;
+    currentPtr_ = nullptr;
+  }
+
+  // The CPU access memory with the alignment,
+  // So construct object from alignment address will reduce the CPU access count then
+  // speed up read/write
+  void *allocateAligned(const std::size_t alloc);
```

Comment:
Consider only provide an interface of `allocate`. It should not be caller's duty to provide an aligned size. When user call `allocate(not aligned size)`, you can round up the size and call `allocateAligned`.

Or provide both `allocateAligned` and `allocate` LGTM as well.

Reply:
The user don't need to provide aligned size, the method only accept one parameter which is allocated size.

Reply:
👌 , my bad